string file_path
---
string[] directory_names
bool result
